### ISSUE TITLE
Drag and drop of media files not allowed on paid simple sites

### ISSUE THREAD

**Issue by gikaragia** [Automattic/wp-calypso#93778]
### Quick summary

Trying to drag and drop media files to the editor, it fails.

### Steps to reproduce

1. Create a site with the Personal or Premium plans
2. Open the page to create a new post
3. Try to drag&drop an audio file (e.g. .mp3 or .wav)

### What you expected to happen

The audio file to be uploaded and the audio blog to be created normally

### What actually happened

The audio file is not uploaded and a message of 'Sorry, you are not allowed to upload this file type.'

### Impact

All

### Available workarounds?

Yes, difficult to implement

### If the above answer is "Yes...", outline the workaround.

The user can still go to the media library, upload the file, and then select it in the audio block.

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

I did some debugging and it seems that the error is caused by the function `wpcom_site_has_upgraded_upload_filetypes` (and `get_current_blog_id`) retrieving the wrong `blog_id`. It retrieves the `blog_id` of the API and not the `blog_id` that the user is currently modifying.

Subsequent requests and calls to the same method retrieve the correct `blog_id` which hints that the method in that case is either called too early or the `blog_id` is not correctly set.

**Comment by jartes** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Replicated
- Tested on Atomic – Replicated

📌 FINDINGS/SCREENSHOTS/VIDEO
- I can replicate this unproxied, and on Atomic sites. 
- While it works for mp3 files, it doesn't work for wav and ogg files.

https://github.com/user-attachments/assets/3b4a5510-b36d-49c7-888f-af9d4bd081b5

📌 ACTIONS
- Triaged

**Comment by gikaragia** [comment]
I tried this again and it works for me now too. It was happening consistently before. I'll try again tomorrow to see if I can reproduce it and let you know of the results.

**Comment by gikaragia** [comment]
Gave it another go and now it happens again:
![php](https://github.com/user-attachments/assets/d69ea9b7-9c8c-40f3-a6ce-093500086d66)

Here is the file that I am trying to upload: https://tinyurl.com/24tkan96

And the blog id: 220057576

**Comment by DustyReagan** [comment]
I can consistently reproduce this for WAV files on Firefox, but Chrome works fine for me. @gikaragia, could you please check if you're experiencing the same behavior?

**Comment by Imran92** [comment]
I also couldn't reproduce it on chrome in a personal plan

**Comment by gikaragia** [comment]
Indeed @DustyReagan this is currently only happening in Firefox and with .wav files (although when I raised I remember it happening with other file types too). Sorry, didn't cross my mind to test with a different browser.

**Comment by DustyReagan** [comment]
Thanks, @Imran92 and @gikaragia! I also see this issue on Atomic and, more significantly, Pressable sites in the core (no Calypso). When I try to upload a `.wav` file, I see a console error `Security Error: Content at [redacted]. may not load data from blob:[redacted].` I've tried reducing my Firefox security settings but still can't get the upload to work. I'm guessing this has something to do with uploading a `blob` and maybe Content Security Policy settings. I think this may be a Guttenberg issue. 🤔

**Comment by Imran92** [comment]
I think I've found the reason and the fix.

It involves a weird behavior of Firefox. When we upload the `wav` file @gikaragia gave us [here](https://github.com/Automattic/wp-calypso/issues/93778#issuecomment-2306885584), Chrome and all other browsers identify the `mimetype` of the file to be `audio/wav`, but Firefox, identifies the `mimetype` as `audio/x-wav`.

WordPress core has a default (extensible) set of mime-type that it allows uploading. You can find them here https://github.com/WordPress/wordpress-develop/blob/3b1e57752d5dd737477873b6797e3f8209aec078/src/wp-includes/functions.php#L3392-L3513 . You’ll notice that there is the `audio/wav` mime-type there, but not `audio/x-wav`.
So when Gutenberg tries to check the `mimetype` of the file in Firefox here https://github.com/WordPress/gutenberg/blob/5773ae60ca33bda03f2a35e6707d961b8db3af18/packages/media-utils/src/utils/upload-media.js#L91-L118, it sees that the mime-type (`audio/x-wav`) isn’t one of the allowed ones.
So it throws an error.

So, I think the proper fix will be to add the audio/x-wav in the wp-core function.

Also, an workaround will be using the mime_types filter. For example, as POC, you can try adding the following snippet and your wav file upload should start working as expected on Firefox

```
add_filter( 'mime_types', function( $mime_types ) {
    $mime_types['wavx'] = 'audio/x-wav';

    return $mime_types;
} );
```

**Comment by miksansegundo** [comment]
I tested the fix and works nicely. Nice find!

The fix is just a small configuration change on the mime file types supported in WordPress core. @annezazu, maybe you can help coordinate with core folks to get it merged?

**Comment by annezazu** [comment]
On it :)

**Comment by davemart-in** [comment]
Removing the Groundskeeper label since this seems to have a fix.

**Comment by ramonjd** [comment]
Gutenberg:

https://github.com/WordPress/gutenberg/pull/66850


Core:

https://github.com/WordPress/wordpress-develop/pull/7265

**Comment by taipeicoder** [comment]
Summing up the state of the issue: 
- The fix is already available in Gutenberg 19.7 via https://github.com/WordPress/gutenberg/pull/66850.
- The fix will be available in the upcoming WordPress 6.8 via https://core.trac.wordpress.org/changeset/59389.

Let's close this issue since its title only refers to Simple sites, and the issue has been fixed for Simple and Atomic (with Gutenberg activated) sites.
